class-vip-support-user.php

```diff
@@ -470,6 +471,26 @@ public function filter_vip_support_email_gravatars( $url, $id_or_email, $args )
 		return $url;
 	}
 
+	/**
+	 * Don't show the admin email confirmation screen to vip support users.
+	 *
+	 * @param string           $redirect_to The redirect destination URL.
+	 * @param string           $requested   The requested redirect destination URL passed as a parameter.
+	 * @param WP_User|WP_Error $user        WP_User object if login was successful, WP_Error object otherwise.
+	 *
+	 * @return $redirect_to
+	 */
+	public function disable_admin_email_check( $redirect_to, $requested, $user ) {
+		if ( isset( $user->ID ) && $this->has_vip_support_meta( $user->ID ) ) {
+			// This filter doesn't always have access to the user object, that's why the above wrapping filter is needed.
```

Comment:
This filter being `login_redirect`? Or `admin_email_check_interval`?

Feels like it would be better to just hook into `admin_email_check_interval` and do the user check there (rather than the double-filter). Would make it easier to add unit tests as well, which we should add for this.

Reply:
It's `admin_email_check_interval` that won't know the current user. This all happens very early in the sign-in process unfortunately, even before error processing in fact.

Only `wp_signon()` is called before this point, and it doesn't handle setting the current user: https://github.com/WordPress/WordPress/blob/f93ee2ca76164bcae721e4730c92ee1455fa1dd9/wp-includes/user.php#L20

To get the user directly in `admin_email_check_interval`, we'd need to check $_POST variables / cookies and run the authentication over again, which is not a great idea. The option of passing `$user` to this filter in core also isn't great, since the other place this filter is used definitely won't have access to the current user - it just checks a nonce and finishes the request, no auth at all.

Reply:
Hey @mjangda, just checking back in on this.

Looked into it some more and I'm really not seeing any other ways to go about this short of WP core reworking the login flow.